admin/agentlocal/agentlocal.go

```diff
@@ -31,21 +33,57 @@ import (
 	agentlocal "github.com/percona/pmm/api/agentlocalpb/json/client/agent_local"
 )
 
+// DefaultClient is http client configured either for socket or tcp connection to local pmm-agent.
+// It shall be used to talk to the local pmm-agent.
+var DefaultClient = http.DefaultClient
+
 // SetTransport configures transport for accessing local pmm-agent API.
-func SetTransport(ctx context.Context, debug bool, port uint32) {
+func SetTransport(ctx context.Context, debug bool, port uint32, socket string) {
 	// use JSON APIs over HTTP/1.1
-	transport := httptransport.New(fmt.Sprintf("%s:%d", Localhost, port), "/", []string{"http"})
+	transport := configureTransport(port, socket)
```

Comment:
I assume that this code can be simplified

```
transport := httptransport.New(fmt.Sprintf("%s:%d", Localhost, port), "/", []string{"http"}) // default fallback transport

if socket != "" {
		transport =  &http.Transport{
			DialContext: func(ctx context.Context, _, _ string) (net.Conn, error) {
                                 var dialer net.Dialer
				return dialer.DialContext(ctx, "unix", socket)
			},
		},
}
```

Reply:
`httptransport.New` returns `*httptransport.Runtime` while you would try to assign `*http.Transport` later. These are incompatible types I believe.